librz/io/io_memory.c

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: LGPL-3.0-only
```

Comment:
This whole refactoring is great! Though probably this entire thing can be replaced with a simple `RzBuffer` (`rz_buf_new_with_bytes`)